Editorial pass (#268)
ietf-wg-privacypass/base-drafts: draft-ietf-privacypass-protocol.md

{::boilerplate bcp14}

> The following terms are used throughout this document. 

Reviewer: ```suggestion
The following terms are used throughout this document.
```

> - Client: Clients obtain Tokens from an Issuer and redeem them for authorization.

Reviewer: ```suggestion
- Client: An entity that runs the Issuance protocol with an Issuer to produce Tokens that can be later used for redemption (see {{Section 2.2 of AUTHSCHEME}}).
```

                   \------------------------------------/
~~~

Issuers provide a Private and Public Key, denoted `skI` and `pkI` respectively,
used to produce tokens as input to the protocol. See {{issuer-configuration}}
for how this key-pair is generated.

Clients provide the following as input to the issuance protocol:

> - Issuer name: identifies the Issuer. This is typically a host name that

Reviewer: ```suggestion
- Issuer name: An identifier for the Issuer. This is typically a host name that
```

  can be used to construct HTTP requests to the Issuer.
- Issuer Public Key: `pkI`, with a key identifier `token_key_id` computed as
  described in {{issuer-configuration}}.
- Challenge value: `challenge`, an opaque byte string. For example, this might
  be provided by the redemption protocol in {{AUTHSCHEME}}.

Given this configuration and these inputs, the two messages exchanged in
this protocol are described below. This section uses notation described in
{{OPRF, Section 4}}, including SerializeElement and DeserializeElement,
SerializeScalar and DeserializeScalar, and DeriveKeyPair.

> `Ne` and `Ns` are as defined in {{OPRF, Section 4}} for OPRF(P-384, SHA-384).

Reviewer: ```suggestion
The constants `Ne` and `Ns` are as defined in {{OPRF, Section 4}} for OPRF(P-384, SHA-384).
```

- "token_type" is a 2-octet integer, which matches the type in the challenge.

- "truncated_token_key_id" is the least significant byte of the `token_key_id` ({{issuer-configuration}}) in network byte order (in other words, the last 8 bits of `token_key_id`).

- "blinded_msg" is the Ne-octet blinded message defined above, computed as

>   `SerializeElement(blinded_element)`. 

Reviewer: ```suggestion
  `SerializeElement(blinded_element)`.
```

## Token Verification

> Tokens are verified by creating a VOPRF context using the Issuer Private Key and Public Key,

Reviewer: ```suggestion
Verifying a Token requires creating a VOPRF context using the Issuer Private Key and Public Key,
```

Clients provide the following as input to the issuance protocol:

> - Issuer name: identifies the Issuer. This is typically a host name that

Reviewer: ```suggestion
- Issuer name: An identifier for the Issuer. This is typically a host name that
```

## Token Verification

> Tokens are verified by checking that Token.authenticator is a valid

Reviewer: ```suggestion
Verifying a Token requires checking that Token.authenticator is a valid
```
